### PR TITLE
Add min/max options to IntegerField

### PR DESCRIPTION
**Pull request by Swahvay**
As the title says. Mostly just copied from `StringField`.

### CODE REVIEW

File: ts/src/schema/field.ts [hunk 1]
```diff
@@ -98,11 +95,65 @@ export function UUIDType(options: FieldOptions): UUIDField {
...
+
+  constructor(options?: IntegerOptions) {
+    super();
+    // for legacy callers
+    this.handleOptions(options || { name: "field" });
+  }
+
+  getOptions(): IntegerOptions {
+    return this.getOptions;
+  }
```
Comment (Swahvay) [inline ts/src/schema/field.ts:115]: Curious what this is for.

File: ts/src/schema/field.ts [hunk 2]
```diff
@@ -98,11 +95,65 @@ export function UUIDType(options: FieldOptions): UUIDField {
...
+  validate(validator: (str: number) => boolean): IntegerField {
+    this.validators.push(validator);
+    return this;
+  }
 }
 
-export function IntegerType(options: FieldOptions): IntegerField {
+export function IntegerType(options: IntegerOptions): IntegerField {
   let result = new IntegerField();
   return Object.assign(result, options);
```
Comment (Swahvay) [inline ts/src/schema/field.ts:158]: Also not totally sure how this works. Why is it not just `return new IntegerField(options)`?

File: ts/src/schema/field.ts (hunk 1 above)
Comment (lolopinto) [reply ts/src/schema/field.ts:115]: it's used at https://github.com/lolopinto/ent/blob/main/ts/src/schema/string_field.test.ts#L276

File: ts/src/schema/field.ts (hunk 2 above)
Comment (lolopinto) [reply ts/src/schema/field.ts:158]: I want the options to be part of the field's properties. used in codegen at https://github.com/lolopinto/ent/blob/main/ts/src/parse_schema/parse.ts#L13

**Comment by lolopinto** [comment]
can you add a simple test file similar to string_field.test.ts?

**Review by lolopinto — CHANGES_REQUESTED** [review]
for the test file

File: ts/src/schema/field.ts (hunk 1 above)
Comment (Swahvay) [reply ts/src/schema/field.ts:115]: Still confused about this part though. `getOptions` never seems to actually be defined anywhere. How does this return anything?

Comment (lolopinto) [reply ts/src/schema/field.ts:115]: got it. i see what you mean. this seems like a very weird bug. not sure what happened here.

```
diff --git a/ts/src/schema/field.ts b/ts/src/schema/field.ts
index 29b35802..5066f75b 100644
--- a/ts/src/schema/field.ts
+++ b/ts/src/schema/field.ts
@@ -151,6 +151,7 @@ export class StringField extends BaseField implements Field {
   type: Type = { dbType: DBType.String };
   private validators: { (str: string): boolean }[] = [];
   private formatters: { (str: string): string }[] = [];
+  private options: StringOptions = { name: "field" };
 
   constructor(options?: StringOptions) {
     super();
@@ -159,7 +160,7 @@ export class StringField extends BaseField implements Field {
   }
 
   getOptions(): StringOptions {
-    return this.getOptions;
+    return this.options;
   }
 
   private handleOptions(options: StringOptions) {
@@ -190,6 +191,7 @@ export class StringField extends BaseField implements Field {
         noParams[k].apply(this);
       }
     }
+    this.options = options;
   }
 
   minLen(l: number): StringField {
diff --git a/ts/src/schema/string_field.test.
```

it should have been something like this

File: ts/src/schema/integer_field.test.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      valid: false,
+    },
+  ],
+]);
+
+for (const [k, v] of tests) {
+  test(k, () => {
+    const f = v.fn();
+    let val = v.customInput || input;
+    expect(f.format(val)).toBe(v.format);
```
Comment (lolopinto) [inline ts/src/schema/integer_field.test.ts:98]: there's no formatting on integer so you should kill this

File: ts/src/schema/integer_field.test.ts [hunk 4]
```diff
@@ -0,0 +1,117 @@
+import { IntegerField, IntegerType, ListField } from "./field";
+
+interface testCase {
+  fn: () => IntegerField;
+  format: number;
```
Comment (lolopinto) [inline ts/src/schema/integer_field.test.ts:5]: kill also

File: ts/src/schema/integer_field.test.ts [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+  for (const [k, v] of tests) {
+    if (k.indexOf("property") === -1) {
+      continue;
+    }
+    test(k, () => {
+      const f = v.fn();
+      const list = new ListField(f, f.getOptions());
+      let val = v.customInput || input;
+      // list format for postgres list
+      expect(list.format([val])).toBe(`{${v.format}}`);
```
Comment (lolopinto) [inline ts/src/schema/integer_field.test.ts:113]: this i think is fine

**Review by lolopinto — CHANGES_REQUESTED** [review]
if you npm|yarn install and then run `jest src/schema/integer_field.test.ts` you should be able to run this test locally

File: ts/src/schema/field.ts (hunk 1 above)
Comment (lolopinto) [reply ts/src/schema/field.ts:115]: https://github.com/lolopinto/ent/pull/590

**Pull request merged**